sirhelpers.h

```diff
@@ -64,32 +64,46 @@ uint16_t _sir_geterrcode(uint32_t err) {
 
 /** Even more evil macros used for binary searching arrays. */
 # define _SIR_DECLARE_BIN_SEARCH(low, high) \
-    size_t _low  = low; \
-    size_t _high = high; \
+    size_t _low  = low;                     \
+    size_t _high = high;                    \
     size_t _mid  = (_low + _high) / 2;
 
 # define _SIR_BEGIN_BIN_SEARCH() do {
-# define _SIR_ITERATE_BIN_SEARCH(comparison) \
-    if (0 == comparison) { \
-        break; \
-    } \
-    \
-    if (_low == _high) \
-        break; \
-    \
+# define _SIR_ITERATE_BIN_SEARCH(comparison)    \
+    if (0 == comparison) {                      \
+        break;                                  \
+    }                                           \
+                                                \
+    if (_low == _high)                          \
+        break;                                  \
+                                                \
     if (0 > comparison && (_mid - 1) >= _low) { \
-        _high = _mid - 1; \
-    } else if ((_mid + 1) <= _high) { \
-        _low = _mid + 1; \
-    } else { \
-        break; \
-    } \
-    \
+        _high = _mid - 1;                       \
+    } else if ((_mid + 1) <= _high) {           \
+        _low = _mid + 1;                        \
+    } else {                                    \
+        break;                                  \
+    }                                           \
+                                                \
     _mid = (_low + _high) / 2;
-
 # define _SIR_END_BIN_SEARCH() \
     } while (true);
 
+/** Validates a pointer and fails if it's invalid. */
+# define _sir_validptr(p) __sir_validptr((const void* restrict)p, true)
+
+/** Validates a pointer-to-pointer and fails if it's invalid. */
+# define _sir_validptrptr(pp) __sir_validptrptr((const void* restrict*)pp, true)
+
+/**
+ * Validates a pointer-to-pointer, pointer,
+ * pointer to function, etc. but ignores whether it's invalid.
+ *
+ * This is necessary due to the fact that ::_sir_validptr will
+ * not accept these types as input.
+ */
+# define _sir_notnull(addr) (NULL != (addr))
```

Comment:
This should have been deleted; if it's in use anywhere it should not be, and should be replaced with one of:

1. _sir_validptr
2. _sir_validptrptr
3. _sir_validfnptr

Reply:
Fixed.